src/traverse/Binding.ts

```diff
@@ -562,6 +554,18 @@ export class VariableBinding extends Binding {
         this.nullifyingPaths[0].parentPath.parent.rightHandSide;
       this.updateOwnership(owner, 'value');
     }
+    this.nullifyingPaths.forEach(path => {
+      const functionDefScope = path.scope.getAncestorOfScopeType(
+        'FunctionDefinition',
+      );
+      if (functionDefScope.callerRestriction === 'match') {
+        this.updateOwnership(functionDefScope.callerRestrictionNode);
+        return;
+      }
+      if (functionDefScope.callerRestriction === 'exclude') {
+        this.updateBlacklist(functionDefScope.callerRestrictionNode);
+      }
+    }); 
```

Comment:
can you please leave a comment explaining these if-statements.

Reply:
I've added the comment.